### ISSUE TITLE
libvirt is broken on x86_64-darwin

### ISSUE THREAD

**Issue by jakobrs** [NixOS/nixpkgs#89166]
**Describe the bug**
`libvirt` is broken on x86_64-darwin, probably because `rpcgen` is missing.

```
  GEN      logging/log_protocol.h
Can't exec "no": No such file or directory at ../../src/rpc/genprotocol.pl line 45.
cannot run no -h ../../src/logging/log_protocol.x: No such file or directory at ../../src/rpc/genprotocol.pl line 45.
```

`libvirt` builds fine when build using `nix-shell` *without `--pure`*. Line 45 of `src/rpc/genprotocol.pl` appears to attempt to invoke `rpcgen`:

```perl
if ($rpcgen =~ /portable-rpcgen/) {
    $rpcgen = "$rpcgen -o -";
}
open RPCGEN, "-|", "$rpcgen $mode $xdrdef"    # This is line 45
    or die "cannot run $rpcgen $mode $xdrdef: $!";
open TARGET, ">$target"
    or die "cannot create $target: $!";
```

Instead of invoking `rpcgen`, `genprotocol` attempts to invoke `no` instead. This is probably because `rpcgen` is not found by `configure`:

```diff
$ diff no-pure-configurePhase.log pure-configurePhase.log 
64c64
< configure flags: --disable-static --prefix=/nix/store/c4qf09sfjgdqz3x796jzr0b8jkv78jgf-libvirt-6.2.0 --prefix=/nix/store/c4qf09sfjgdqz3x796jzr0b8jkv78jgf-libvirt-6.2.0 --disable-static --prefix=/nix/store/c4qf09sfjgdqz3x796jzr0b8jkv78jgf-libvirt-6.2.0 --localstatedir=/var --sysconfdir=/var/lib --with-libpcap --with-qemu --with-vmware --with-vbox --with-test --with-esx --with-remote --with-polkit --with-init-script=none
---
> configure flags: --disable-static --prefix=/nix/store/c4qf09sfjgdqz3x796jzr0b8jkv78jgf-libvirt-6.2.0 --localstatedir=/var --sysconfdir=/var/lib --with-libpcap --with-qemu --with-vmware --with-vbox --with-test --with-esx --with-remote --with-polkit --with-init-script=none
148c148
< checking for nmedit... nmedit
---
> checking for nmedit... no
531c531,532
< checking for rpcgen... /usr/bin/rpcgen   # See this line
---
> checking for rpcgen... no                # and this line
> checking for portable-rpcgen... no
```

However, `libvirt-5.4.0` built fine and the build logs for `libvirt-5.4.0` also say that there's no `rpcgen`.

It appears that version 5.4.0 had "prebuilt" the relevant files, and that `src/rpc/genprotocol.pl` therefore wasn't actually being run:
```
$ ls libvirt-5.4.0/src/logging/log_protocol.*
libvirt-5.4.0/src/logging/log_protocol.c  libvirt-5.4.0/src/logging/log_protocol.h  libvirt-5.4.0/src/logging/log_protocol.x
$ ls libvirt-6.2.0/src/logging/log_protocol.*
libvirt-6.2.0/src/logging/log_protocol.x
```

I've tried to add both `netbsd.rpcgen` and `darwin.developer_cmds` to `buildInputs`, and neither worked.

---

After some more analysis:
- The globally-installed `/usr/bin/rpcgen` works
- `rpcgen` from `netbsd.rpcgen` produces files that are syntactically correct, but refer to nonexistent/undeclared types
- `rpcgen` from `darwin.developer_cmds` fails due to a syntax error in the `*.x` files

Note that `/usr/bin/rpcgen` and `rpcgen` from `netbsd.rpcgen` produce different files.

[Configure and build logs (without `--pure`)](https://github.com/NixOS/nixpkgs/files/4705283/libvirt-6.2.0-logs.tar.gz) I also have the entire build folders but they're larger than 10MB.

**Notify maintainers**
@fpletz @globin 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.3.0, macOS 10.15.3`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.3`
 - channels(jakobskaar): `"nixpkgs-20.09pre227522.d4226e3a4b5"`
 - nixpkgs: `/Users/jakobskaar/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: nixpkgs.libvirt
# a list of nixos modules affected by the problem
module:
```

**Comment by jakobrs** [comment]
After some more analysis:
- The globally-installed `/usr/bin/rpcgen` works
- `rpcgen` from `netbsd.rpcgen` produces files that are syntactically correct, but refer to nonexistent/undeclared types
- `rpcgen` from `darwin.developer_cmds` fails due to a syntax error in the `*.x` files

Note that `/usr/bin/rpcgen` and `rpcgen` from `netbsd.rpcgen` produce different files.

[Configure and build logs (without `--pure`)](https://github.com/NixOS/nixpkgs/files/4705283/libvirt-6.2.0-logs.tar.gz) I also have the entire build folders but they're larger than 10MB.

**Comment by LuyaoZhong** [comment]
I also met this issue on Fedora 31. Thanks for your workaroud, @jakobrs

**Comment by princeofdream** [comment]
where can i find rpcgen to install...?

**Comment by jakobrs** [comment]
This appears to have been fixed (presumably by #95089, specifically 24fd6016405bdda1406ed644db5be16b7f597f35).
